editor/src/core/shared/property-path.ts

```diff
@@ -52,7 +52,20 @@ function getPathCache(elements: Array<PropertyPathPart>): PropertyPathCache {
   return workingPathCache
 }
 
-export function create(elements: Array<PropertyPathPart>): PropertyPath {
+export function create<T1 extends PropertyPathPart>(element1: T1): PropertyPath<[T1]>
```

Comment:
Does it need to be generic?

Reply:
yeah, this is what makes Typescript infer the narrowest type possible for the individual path elements

<img width="392" alt="image" src="https://user-images.githubusercontent.com/2226774/214875528-10110a0a-e780-4f5a-920e-f373cf39fa17.png">

otherwise it would just infer this, which is not sufficient:
<img width="461" alt="image" src="https://user-images.githubusercontent.com/2226774/214875562-b21fbf43-2e06-4582-a7a7-1a48041c24cd.png">